File: Medium/Java/324. Wiggle Sort II.java
class Solution {
    public void wiggleSort(int[] nums) {
        Arrays.sort(nums);
        int mid = (nums.length - 1) / 2;
        int end = nums.length - 1;
        int[] store = new int[end + 1];
        int count = 0;
        
        while(0 <= mid || (nums.length - 1) / 2 < end){
            if(count % 2 == 0){
                store[count] = nums[mid];
                mid--;
            }else{
                store[count] = nums[end];
                end--;
            }
            count++;
        }

        for(int i = 0; i < nums.length; i++){
            nums[i] = store[i];
        }
    }
}

File: Medium/Python/324. Wiggle Sort II.py
class Solution(object):
    def wiggleSort(self, nums):
        nums.sort()
        mid = (len(nums) - 1) / 2
        end = len(nums) - 1
        store = [0] * (end + 1)
        count = 0

        while 0 <= mid or (len(nums) - 1) / 2 < end:
            if count % 2 == 0:
                store[count] = nums[mid]
                mid -= 1
            else:
                store[count] = nums[end]
                end -= 1
            count += 1
        
        for i in range(len(nums)):
            nums[i] = store[i]
